component-catalog-connectors/artifactory-connector/artifactory_catalog_connector/artifactory_catalog_connector.py

```diff
@@ -0,0 +1,293 @@
+                repository_name=repository_name,
+                file_filter=file_filter,
+                file_ordering=file_ordering,
+                max_recursion_depth=max_recursion_depth,
+                max_files_per_folder=max_files_per_folder,
+                current_folder_path=absolute_folder_path,
+                current_recursion_depth=current_recursion_depth + 1,
+            )
+
+    return component_list
+
+
+class ArtifactoryComponentCatalogConnector(ComponentCatalogConnector):
+    """
+    Read component definitions from an Artifactory catalog
+    """
+
+    def get_catalog_entries(
+        self, catalog_metadata: Dict[str, Any]
+    ) -> List[Dict[str, Any]]:
+        """
+        Returns a list of component_metadata instances, one per component found in the given registry.
+        The form that component_metadata takes is determined by requirements of the reader class.
+
+        :param catalog_metadata: the dictionary-form of the Metadata instance for a single registry
+        """
+        component_list = []
+
+        artifactory_url = catalog_metadata["artifactory_url"]
+        artifactory_username = catalog_metadata.get("artifactory_username")
+        artifactory_password = catalog_metadata.get("artifactory_password")
+        repository_name = catalog_metadata["repository_name"]
+        repository_path = catalog_metadata["repository_path"]
+        max_recursion_depth = int(catalog_metadata["max_recursion_depth"])
+        max_files_per_folder = int(catalog_metadata["max_files_per_folder"])
+        file_filter = catalog_metadata["file_filter"]
+        file_ordering = catalog_metadata["file_ordering"]
+
+        url_obj = urlparse(artifactory_url)
+        api_base_url = f"{url_obj.scheme}://{url_obj.netloc}/{url_obj.path.strip('/')}"
+
+        api_auth = None
+        if artifactory_username and artifactory_password:
+            api_auth = HTTPBasicAuth(
+                username=artifactory_username, password=artifactory_password
+            )
+        elif artifactory_password:
+            api_auth = ArtifactoryApiKeyAuth(api_key=artifactory_password)
+
+        try:
+            self.log.debug("Retrieving component list from Artifactory catalog.")
+
+            component_list += recursively_get_components(
+                api_base_url=api_base_url,
+                api_auth=api_auth,
+                repository_name=repository_name,
+                file_filter=file_filter,
+                file_ordering=file_ordering,
+                max_recursion_depth=max_recursion_depth,
+                max_files_per_folder=max_files_per_folder,
+                current_folder_path=repository_path,
+                current_recursion_depth=0,
+            )
+
+        except Exception as ex:
+            self.log.error(
+                f"Error retrieving component list from Artifactory catalog: {ex}"
+            )
+
+        return component_list
+
+    def read_catalog_entry(
```

Comment:
The interface name and definition has changed in version 3.7/master to support return of additional metadata:

```
    def get_entry_data(self,
                       catalog_entry_data: Dict[str, Any],
                       catalog_metadata: Dict[str, Any]) -> Optional[EntryData]:
```